dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1081,6 +1060,21 @@ private void fillInMetadata(DepositMetadata depositMetadata, Workflow workflow,
         // The Zenodo API requires at description of at least three characters
         String descriptionStr = (description == null || description.isEmpty()) ? "no description" : workflow.getDescription();
         depositMetadata.setDescription(descriptionStr);
+
+        // The Zenodo API requires an access right
+        DepositMetadata.AccessRightEnum accessRight = (depositMetadata.getAccessRight() == null)
+                ? DepositMetadata.AccessRightEnum.OPEN : depositMetadata.getAccessRight();
+        depositMetadata.setAccessRight(accessRight);
+
+        // The Zenodo API requires a license type if access right is 'open' or 'embargoed'
+        // We only set the access right to OPEN at this time, and the default license to cc-by
+        // for the metadata (non-datasets)
+        if (accessRight == DepositMetadata.AccessRightEnum.OPEN || accessRight == DepositMetadata.AccessRightEnum.EMBARGOED) {
+            String license = (depositMetadata.getLicense() == null || depositMetadata.getLicense().isEmpty())
+                    ? "CC-BY-4.0" : depositMetadata.getLicense();
```

Comment:
I don't think we should do this -- it seems dicey to arbitrarily assign a license to creative work.

Where does the potential existing value come from? 

Is it possible to get a license from the workflow? Probably not, but just brainstorming.

Does the API force you to have a license? Although I'm not sure that not specifying a license when there is one is a good thing either.

Reply:
I wonder what GitHub+Zenodo does in the case of repos without licenses.

Reply:
Removed code setting license since Zenodo provides a default and the user can edit this metadata on Zenodo

Their API documentation https://developers.zenodo.org/#representation says:

access_right
string | required: Yes | Controlled vocabulary:* open: Open Access* embargoed: Embargoed Access* restricted: Restricted Access* closed: Closed 
Access Defaults to open.

license
string | required: Yes, if access_rightis "open" or "embargoed". | Controlled vocabulary:The selected license applies to all files in this deposition, but not to the metadata which is licensed under Creative Commons Zero. Further information about licenses is available at Open Definition Licenses Service. 
Defaults to cc-by for non-datasets and cc-zero for datasets.